include/dali/core/unique_handle.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_UNIQUE_HANDLE_H_
+#define DALI_CORE_UNIQUE_HANDLE_H_
+
+#include <utility>
+#include "dali/core/common.h"
+
+namespace dali {
+
+template <typename HandleType, typename Actual>
+class UniqueHandle {
+ public:
+  using handle_type = HandleType;
+
+  constexpr inline UniqueHandle() : handle_(Actual::null_handle()) {}
```

Comment:
[This](https://en.cppreference.com/w/cpp/language/constexpr) says, that `constexpr` implies `inline`. Isn't it redundant?

Reply:
I remember having seen speed difference.